python/ray/serve/_private/http_proxy.py

```diff
@@ -72,6 +74,62 @@
     )
 
 
+async def _handle_streaming_response(
+    asgi_response_generator: "ray._raylet.StreamingObjectRefGenerator",
+    scope: Scope,
+    receive: Receive,
+    send: Send,
+) -> str:
+    """Consumes the `asgi_response_generator` and sends its data over `send`.
+
+    This function is essentially a proxy for a downstream ASGI response. The
+    passed generator is expected to return a stream of pickled ASGI messages
+    (dictionaries) that will be passed to the provided ASGI send interface.
+
+    Exception handling depends on if the first message has already been sent:
+        - if an exception happens *before* the first message, a 500 status will be sent.
+        - if an exception happens *after* the first message, the response stream will be
+          terminated.
+
+     This is because once the first message has been sent, the client has already
```

Comment:
I don't follow this sentence because I don't follow what "this" is. Does this apply to the first, second, or both exception handling scenarios?

Reply:
makes sense. updated to:

```
 94     The difference in behavior is because once the first message has been sent, the
 95     client has already received the status code so we cannot send a `500` (internal
 96     server error).
```